neo/SmartContract/Native/Tokens/NeoToken.cs

```diff
@@ -0,0 +1,422 @@
+                        storage_count.Value = state_count.ToByteArray();
+                    }
+                }
+            }
+            engine.Service.SendNotification(engine, ScriptHash, new StackItem[] { "Transfer", from, to, amount });
+            return true;
+        }
+
+        private static void DistributeGas(ApplicationEngine engine, byte[] account, AccountState state)
+        {
+            BigInteger gas = CalculateBonus(engine, state.Balance, state.BalanceHeight, engine.Service.Snapshot.PersistingBlock.Index);
+            state.BalanceHeight = engine.Service.Snapshot.PersistingBlock.Index;
+            GasToken.DistributeGas(engine, account, gas);
+        }
+
+        private static BigInteger CalculateBonus(ApplicationEngine engine, BigInteger value, uint start, uint end)
+        {
+            if (value.IsZero || start >= end) return BigInteger.Zero;
+            if (value.Sign < 0) throw new ArgumentOutOfRangeException(nameof(value));
+            uint amount = 0;
+            uint ustart = start / Blockchain.DecrementInterval;
+            if (ustart < Blockchain.GenerationAmount.Length)
+            {
+                uint istart = start % Blockchain.DecrementInterval;
+                uint uend = end / Blockchain.DecrementInterval;
+                uint iend = end % Blockchain.DecrementInterval;
+                if (uend >= Blockchain.GenerationAmount.Length)
+                {
+                    uend = (uint)Blockchain.GenerationAmount.Length;
+                    iend = 0;
+                }
+                if (iend == 0)
+                {
+                    uend--;
+                    iend = Blockchain.DecrementInterval;
+                }
+                while (ustart < uend)
+                {
+                    amount += (Blockchain.DecrementInterval - istart) * Blockchain.GenerationAmount[ustart];
+                    ustart++;
+                    istart = 0;
+                }
+                amount += (iend - istart) * Blockchain.GenerationAmount[ustart];
+            }
+            amount += (uint)(engine.Service.Snapshot.GetSysFeeAmount(end - 1) - (start == 0 ? 0 : engine.Service.Snapshot.GetSysFeeAmount(start - 1)));
+            return value * amount * GasToken.DecimalsFactor / TotalAmount;
+        }
+
+        private static bool Initialize(ApplicationEngine engine)
+        {
+            if (engine.Service.Trigger != TriggerType.Application) throw new InvalidOperationException();
+            StorageKey key = CreateStorageKey(ScriptHash, Prefix_Initialized);
+            if (engine.Service.Snapshot.Storages.TryGet(key) != null) return false;
+            engine.Service.Snapshot.Storages.Add(key, new StorageItem
+            {
+                Value = new byte[] { 1 },
+                IsConstant = true
+            });
+            byte[] account = Contract.CreateMultiSigRedeemScript(Blockchain.StandbyValidators.Length / 2 + 1, Blockchain.StandbyValidators).ToScriptHash().ToArray();
```

Comment:
I think that `initialize` should receive this script as parameter, instead of hardcoding it here. This looks more like a Blockchain (ledger) decision than a Neo token decision.